ledger-tool/src/main.rs

```diff
@@ -1146,6 +1218,30 @@ fn main() {
                             "geyser_plugin_config",
                         ])
                         .help("In addition to the bank hash, optionally include accounts and/or transactions details for the slot"),
+                )
+                .arg(
+                    Arg::with_name("abort_on_invalid_block")
+                        .long("abort-on-invalid-block")
+                        .takes_value(false)
+                        .help(
+                            "Exits with failed status early as soon as any bad block is detected",
+                        ),
+                )
+                .arg(
+                    Arg::with_name("no_block_cost_limits")
```

Comment:
think tht practically there is not really a difference - we're not gonna be filling u64::MAX anytime soon.
but the name here is somewhat contradictory with the help.
We still have block-limits they are just u64::MAX-ed, such that they do not "functionally" apply.

I think can either change name here to `max_block_cost_limits`, or adjust the description.

Reply:
> I think can either ..., or adjust the description.

I took this idea: 3cd918d32288550a9f0cf1ad270c965de5f86431


> we're not gonna be filling u64::MAX anytime soon.

Well, I think it's relatively soon with a similar reasoning of lifting the hard-coded 4 banking threads (https://github.com/anza-xyz/agave/pull/2733#issuecomment-2315475660). Once after unified scheduler is enabled across board (v2.1+), block verification is kind of guaranteed to complete faster than block production, assuming num of thread of bp <= num of thread of bv and no bankless/async banking. I'm confirming this via replaying the simed blocks.

so, banking will be aborted by the 350ms cutoff, not by the cu limits. so, the cu limits will work as some kind of the last resort safety guard sitting around 400-500M cu / block.

Reply:
not all CUs are equal since some things are currently mispriced - we'll need to evaluate block limits carefully for those cases, but should also take into consideration the much more efficient replay algorithm you've introduced.

Reply:
> not all CUs are equal since some things are currently mispriced - we'll need to evaluate block limits carefully for those cases

while i'm agreeing with careful review of adjusting them, that's why i want to lower importance to those cu limits than today. cu in general isn't appropriate for the proxy of wall time. and wall time itself can be used, imo.

it's really hard for a given _deterministic_ computation takes substantially longer on a replaying node than a banking node in term of wall time. So, as long as majority of leaders are honest to stop producing blocks after 350ms, i think there's only limited threat to the so called jumbo block problems.

Reply:
> and wall time itself can be used, imo.

But it can't if we move to full or semi asynchronous execution.

> it's really hard for a given individual deterministic computations (ie transactions) to take substantially longer on a replaying node than a banking node

Limits are not fully there for well-behaved nodes, they're primarily there for malicious nodes.

Reply:
> Do you see a potential desire for setting a custom value here different than `u64::MAX` ?

good call. the answer is _to some extent_. 

> If so, we could rename to `block_cost_limit` and allow optional value. So, behavior would be:
> 
>     * Flag passed + no value set ==> set the limit to `u64::MAX`

Actually, i considered this flag. but i didn't think this is intuitive.

If it turns out that there's a strong desire for custom block limits, I'll add a new flag, which conflicts with `--no-block-cost-limit` in the future.